src/dependency-manager/spec/database-spec.ts

```diff
@@ -0,0 +1,37 @@
+import { IsOptional, IsString } from 'class-validator';
+import { JSONSchema } from 'class-validator-jsonschema';
+import { AnyOf, ExpressionOr } from './utils/json-schema-annotations';
+import { Slugs } from './utils/slugs';
+
+export class DatabaseSpec {
+  @IsOptional()
+  @JSONSchema({
+    type: 'string',
+    pattern: Slugs.ArchitectSlugValidator.source,
+    errorMessage: Slugs.ArchitectSlugDescription,
+    description: 'A specific service name which will override the database name specified in the component.',
+  })
+  reserved_name?: string;
+
+  @IsOptional()
+  @JSONSchema({
+    type: 'string',
+    description: 'Human readable description',
+  })
+  description?: string;
+
+  @IsString()
+  @JSONSchema({
+    ...ExpressionOr({ type: 'string', pattern: Slugs.ComponentDatabaseValidator.source }),
+    description: 'The type engine and version of database software needed for data storage.',
+    errorMessage: Slugs.ComponentDatabaseDescription,
+  })
+  type!: string;
```

Comment:
We won't need the engine and version here - I think this should be `engine` and not `type` - `engine` will be more specific and prompt users to enter something like "postgres" or "mysql", while `type` might prompt a user to instead give "SQL" or "NoSQL".

Reply:
Or, to stay more consistent with DatabaseContext, this could be `protocol`

Reply:
I avoided that because I consider the the value to be `<engine>:<version>`. If we pull version into a different field then I think engine makes sense, but I was considering `type` to be the joint value. This also helps us mirror docker tag conventions which aids us in generating dockerized DB instances.

Reply:
I am not sold on type here either because I don't feel it does a good enough job of conveying exactly what the value is on first glance. Though I can't think of something better than type at least right now.

Reply:
I don't think we need to overthink it. The configuration for a database is so minimal that it'll be easy no matter what the keyword is.